workspaces/src/rpc/client.rs

```diff
@@ -0,0 +1,69 @@
+// TODO: Remove this when near-jsonrpc-client crate no longer defaults to deprecation for
+//       warnings about unstable API.
+#![allow(deprecated)]
+
+use tokio_retry::strategy::{jitter, ExponentialBackoff};
+use tokio_retry::Retry;
+
+use near_jsonrpc_client::{methods, JsonRpcClient, JsonRpcMethodCallResult};
+use near_primitives::transaction::SignedTransaction;
+use near_primitives::views::FinalExecutionOutcomeView;
+
+use crate::runtime::context::MISSING_RUNTIME_ERROR;
+
+fn rt_current_addr() -> String {
+    crate::runtime::context::current()
+        .expect(MISSING_RUNTIME_ERROR)
+        .rpc_addr()
+}
+
+fn json_client() -> JsonRpcClient {
+    JsonRpcClient::connect(&rt_current_addr())
+}
+
+pub(crate) fn new() -> RetryClient {
+    RetryClient::new()
+}
+
+pub struct RetryClient;
```

Comment:
does this need to be pub and not crate scoped? I see it doesn't seem to be exported, but might be safe, just in case a refactor exposes this type?
```suggestion
pub(crate) struct RetryClient;
```